Makefile

```
@@ -28,7 +31,7 @@ config-git:
 subctl: config-git
 	./scripts/subctl.sh $(SUBCTL_ARGS)
 
-e2e: $(if $(TARGET_RELEASE),deploy)
+e2e: $(if $(DEPLOY_NEEDED),deploy)
 	./scripts/e2e.sh
```

How about renaming `DEPLOY_NEEDED` to `E2E_NEEDED` and then

```suggestion
ifeq (true, $(E2E_NEEDED))
e2e: deploy
	./scripts/e2e.sh
else
e2e:
endif
```